include/contractor/graph_contractor.hpp

```diff
@@ -101,12 +101,14 @@ class GraphContractor
 
     void Run(double core_factor = 1.0);
 
-    void GetCoreMarker(std::vector<bool> &out_is_core_node);
+    std::vector<bool> GetCoreMarker();
```

Comment:
Thanks for getting rid of the `please fill my vector` way. I feel this way is much better. How do you feel about adding some extra commentary here? As far as I read the code, both vectors get destroyed when accessing here, right?